### PR TITLE
Refactor/ruby browse gem

### LINKED ISSUE #11: :RubyBrowseGem refactor
Code of RubyBrowseGem function is kinda weird
@cuducos

### PR DESCRIPTION
**Pull request by vinibispo**
Close #11

### CODE REVIEW

File: lua/ruby_nvim/cmd.lua [hunk 1]
```diff
@@ -85,24 +84,20 @@ end
...
-  local gem_index = line:find('gem')
-  local gem_with_space_and_quote_length = 5
-  local quote_and_comma_length = 2
-  local comma_index = line:find(',')
-  local gem_name = ''
-  if not gem_index then
-    vim.notify(string.format("There is no gem in this line"), "error")
+  local node = util.get_method_relevant_to_cursor()
+  if node == nil then
+    vim.notify("gem not found", "error")
```
Comment (cuducos) [inline lua/ruby_nvim/cmd.lua]: I think the message here might be misleading. The _gem not found_ looks like, to me, that the plugin has got a name for the Gem, but couldn't find it, e.g. in rubygems.org. What about something like _There is no gem close to the cursor_ to keep it aligned with the other error message?

File: lua/ruby_nvim/util.lua [hunk 2]
```diff
@@ -101,4 +117,51 @@ M.alternate = function(file_path)
...
+  return get_methods(root)
+end
+
+M.get_gem_name = function (node, bufnr)
+  while node ~= nil do
+    if node:type() == 'call' then
+      local method = node:field('method')[1]
+      local method_name = vim.treesitter.query.get_node_text(method, bufnr)
+      if method_name == 'gem' then
+        local method_args = node:field('arguments')[1] --that when using get_node_text has text 'aws-sdk-s3', '~> 1'
```
Comment (cuducos) [inline lua/ruby_nvim/util.lua]: I loved these comments, they clarify a lot for the reader! Can we work on the wording a little bit?

If the result is a text:

```lua
        local method_args = node:field('arguments')[1] -- e.g. get_node_text returns "'aws-sdk-s3', '~> 1'"
```

Or, if it's a table:

```lua
        local method_args = node:field('arguments')[1] -- e.g. get_node_text returns {'aws-sdk-s3', '~> 1'}
```

If you like it, you can follow this pattern in the next lines (BTW, in line 136 there's a typo `s/usin/using`).

File: lua/ruby_nvim/util.lua [hunk 3]
```diff
@@ -101,4 +117,51 @@ M.alternate = function(file_path)
...
+        local gem_node_with_quotes = method_args:child(0) -- that when using get_node_text has text 'aws-sdk-s3'
+        local gem_node = gem_node_with_quotes:child(1) -- that when usin get_node_text has text aws-sdk-s3
+        local gem_name = vim.treesitter.query.get_node_text(gem_node, bufnr)
+        if gem_name ~= "'" then
```
Comment (cuducos) [inline lua/ruby_nvim/util.lua]: Can we avoid having a single quote as a gem name? When does this happen?

**Review by cuducos — APPROVED** [review]
I think it looks really good!

Added minor comments, surely feel free to ignore them all : )

File: lua/ruby_nvim/util.lua (hunk 3 above)
Comment (vinibispo) [reply lua/ruby_nvim/util.lua]: When on gemfile you set
```ruby
gem ''
```

File: lua/ruby_nvim/cmd.lua (hunk 1 above)
Comment (vinibispo) [reply lua/ruby_nvim/cmd.lua]: Ok, I'll fix it

File: lua/ruby_nvim/util.lua (hunk 3 above)
Comment (cuducos) [reply lua/ruby_nvim/util.lua]: Wow… bizarre. I mean: is it useful? Why should we expect `gem ''` in a `Gemfile`?
Anyway. Good to go, I guess.

**Pull request merged**